src/mark.c

```diff
@@ -153,8 +155,15 @@ gboolean r_mark_active(RaucSlot *slot, GError **error)
 	g_free(slot_state->activated_timestamp);
 	now = g_date_time_new_now_utc();
 	slot_state->activated_timestamp = g_date_time_format(now, "%Y-%m-%dT%H:%M:%SZ");
-	slot_state->activated_count++;
 
+	if (r_context()->config->prevent_late_fallback == R_CONFIG_FALLBACK_LOCK_COUNTER) {
```

Comment:
Is there a specific reason for placing it between two `slot_state` variable updates?

Another potential issue might be that if `r_boot_set_primar()` succeeds but `r_boot_set_lock_counter()` fails for some reason, you might end in a system that is locked to the new slot, i.e. which does not provide a fallback anymore. So maybe changing the order of the  `r_boot_set_lock_counter()` and `r_boot_set_primar()` call might make sense here?

This would also be consistent with the order used in `r_mark_good()` below because the locking then surrounds the other bootloader operations.

Reply:
I guess this was changed and discussed in the latest version. Will mark it as 'resolved', though.